### PR TITLE
feat(MOPLAT-314): Artwork view should be a carousel

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [MOPLAT-314]

### Description
Whenever viewing an artwork, it should be possible to swipe to the other artworks in the same context (i.e. artworks by the same artist, in the same show, or in the same album).

### Acceptance criteria
* The carousel should make it possible to swipe horizontally between artwork screens
* Each artwork screen can be scrolled vertically

### Differences from other approaches
This approach is more performant because [react-native-pager-view](https://github.com/callstack/react-native-pager-view) (under the hood it is using the native [Android ViewPager](https://developer.android.com/reference/android/support/v4/view/ViewPager) and the [iOS UIPageViewController](https://developer.apple.com/documentation/uikit/uipageviewcontroller) implementations) is used. Performance improvements are especially noticeable on Android devices

#### iOS
https://user-images.githubusercontent.com/3513494/178563786-724c7233-0c52-43de-a4bb-65a1396f981a.mp4

#### Android
https://user-images.githubusercontent.com/3513494/178566167-9cc0ff21-816d-4a6b-ba83-8fcdf805c8b5.mp4

### PR Checklist

- [x] I tested my changes on **iOS** and **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-314]: https://artsyproduct.atlassian.net/browse/MOPLAT-314?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/sharedUI/molecules/ScrollableScreensView/types.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export type ScrollableScreenEntity = {
```
Comment (pvinis) [inline src/app/sharedUI/molecules/ScrollableScreensView/types.ts:1]: didnt tslint ask to make this an interface?

Comment (dimatretyak) [reply src/app/sharedUI/molecules/ScrollableScreensView/types.ts:1]: Nope

File: src/app/sharedUI/molecules/ScrollableScreensView/Context.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { createContext, useContext } from "react"
+import { ScrollableScreenEntity } from "./types"
+
+export interface ScrollableScreensContextValue {
+  activeScreenIndex: number
+  activeScreen: ScrollableScreenEntity
+}
+
+export const ScrollableScreensContext = createContext(null as any)
```
Comment (pvinis) [inline src/app/sharedUI/molecules/ScrollableScreensView/Context.ts:9]: nice. thanks for doing it this way! if you want, that works too

```suggestion
export const ScrollableScreensContext = createContext(null!)
```

File: src/app/sharedUI/molecules/ScrollableScreensView/LazyScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  shouldRender: boolean
+}
+
+// TODO: Remove this component when we need to request data for all artworks
+export const ScrollableLazyScreen: React.FC<ScrollableLazyScreenProps> = ({
+  screen,
+  shouldRender,
+}) => {
+  const [canMount, setCanMount] = useState(false)
+  const isSelfMounted = useRef(true)
```
Comment (pvinis) [inline src/app/sharedUI/molecules/ScrollableScreensView/LazyScreen.tsx:16]: i would love a comment for this and the line above. whats up with these exactly?

File: src/app/sharedUI/molecules/ScrollableScreensView/Context.ts (hunk 2 above)
Comment (dimatretyak) [reply src/app/sharedUI/molecules/ScrollableScreensView/Context.ts:9]: If we do this, then this warning is displayed

image

Comment (pvinis) [reply src/app/sharedUI/molecules/ScrollableScreensView/Context.ts:9]: weird. ok cool

**Review by pvinis — APPROVED** [review]
nice!

File: src/app/sharedUI/molecules/ScrollableScreensView/LazyScreen.tsx (hunk 3 above)
Comment (dimatretyak) [reply src/app/sharedUI/molecules/ScrollableScreensView/LazyScreen.tsx]: Added small improvements and removed unused code

**Comment by dimatretyak** [comment]
> Are you also getting this issue?

I think the TS changes were not fully indexed (you can go to the `app/sharedUI` and `src/app/routes/HomeTabsNavigationStack.tsx` files and the problem should disappear)

**Review by rajsam003 — APPROVED** [review]
🚀

**Comment by dimatretyak** [comment]
> Have you tried this out with like > 100 artworks? It looks like `react-native-pager-view` handles virtualisation well, but it would be important to make sure.

QAed for David Hockney (who has more than 100 artworks) on an Android device. Everything works smoothly and no visual problems were detected.

https://user-images.githubusercontent.com/3513494/178751010-58cf47e0-0148-4265-a1df-16ac15d821f9.mp4

**Pull request merged**